keel-deliveryconfig-plugin/src/main/kotlin/com/netflix/spinnaker/keel/deliveryconfig/resource/DeliveryConfigHandler.kt

```diff
@@ -35,39 +42,98 @@ class DeliveryConfigHandler(
     "deliveryConfig:${spec.name}"
   )
 
+  private val resourcePersister: ResourcePersister by lazy { resourcePersisterProvider.get() }
```

Comment:
defers the bean resolution from the app context until it is actually used, breaks the circular dependency of resourcePersister needs resource handlers but this is a resource handler and it needs the resourcePersister

Reply:
It kind of feels like the handlers should have access to the resource persister. Maybe there's a better way - emitting some sort of resource event that then gets picked up by the API and persists from there?